crates/bevy_ecs/src/event.rs

```diff
@@ -552,6 +552,12 @@ impl<E: Event> Events<E> {
         events.update();
     }
 
+    /// A [`Condition`] that checks if the event's [`update_system`](Self::update_system)
+    /// needs to run or not.
+    pub fn update_condition(events: Res<Self>) -> bool {
```

Comment:
I think this would read more clearly as `need_updating`?

Reply:
That might work, but the fact this takes a `Res<Self>` doesn't conform with that kind of use. Perhaps we could emulate the `update`/`update_system` split for this, though I'm not a huge fan of putting the system on the type itself.

Reply:
> though I'm not a huge fan of putting the system on the type itself.

Agreed, that pattern here has always struck me as pretty odd and hard to read.

Reply:
> this takes a `Res<Self>`

Maybe `has_updates`? Would line up with `&self -> bool`

```rs
.run_if(Events::<T>::has_updates),
```

Reply:
I extracted both of these so they exist separately from the `Events<T>` type.